docs: accessing private Git repos

max-cx opened the pull request:
<!-- 
Please use one of the following prefixes for the title:
docs: Documentation not including procedures. Engineering review is mandatory.
procedures: Documentation including procedures. Testing procedures is mandatory. Engineering and QE review is mandatory (Engineering can review on behalf of QE). 
chore: Routine, release, tooling, version upgrades.
fix: Fix build, language, links, or metadata.
-->

<!-- Read our [Contribution guide](https://github.com/eclipse/che-docs/blob/master/CONTRIBUTING.adoc) before submitting a PR. -->

## What does this pull request change
Replacing https://github.com/eclipse-che/che-docs/pull/2242, which was for `main`.

## What issues does this pull request fix or reference

## Specify the version of the product this pull request applies to

## Pull request checklist

The author and the reviewers validate the content of this pull request with the following checklist, in addition to the [automated tests](code_review_checklist.adoc).

- Any procedure:
  - [ ] Successfully tested.
- Any page or link rename:
  - [ ] The page contains a redirection for the previous URL.
  - Propagate the URL change in:
    - [ ] Dashboard [default branding data](https://github.com/eclipse-che/che-dashboard/blob/main/packages/dashboard-frontend/src/services/bootstrap/branding.constant.ts)
    - [ ] Chectl [constants.ts](https://github.com/che-incubator/chectl/blob/master/src/constants.ts)
- [ ] Builds on [Eclipse Che hosted by Red Hat](https://workspaces.openshift.com).
- [ ] the *`Validate language on files added or modified`* step reports no vale warnings.

max-cx committed:
modified Spelling.yml

max-cx committed:
modified Spelling.yml

max-cx committed:
Merge remote-tracking branch 'upstream/master'

max-cx committed:
Merge remote-tracking branch 'upstream/master'

max-cx committed:
Merge remote-tracking branch 'upstream/master'

max-cx committed:
Merge remote-tracking branch 'upstream/master'

max-cx committed:
Merge remote-tracking branch 'upstream/master'

max-cx committed:
Merge remote-tracking branch 'upstream/master'

max-cx committed:
Merge remote-tracking branch 'upstream/master'

max-cx committed:
Merge remote-tracking branch 'upstream/master'

max-cx committed:
Merge remote-tracking branch 'upstream/master'

max-cx committed:
draft

max-cx committed:
modified:   antora.yml

max-cx committed:
draft

dkwon17 commented on modules/administration-guide/pages/configuring-authorization-for-private-git-repositories-on-bitbucket.adoc:
```diff
@@ -0,0 +1,21 @@
+To enable Che users to access a private Git repository that is hosted on a Bitbucket server:
+
+. Set up the Bitbucket application link
+. Apply the Bitbucket application link Secret
```
```suggestion
. Apply the Bitbucket application link Secret on the {prod-short} namespace
```

Just a minor suggestion

max-cx replied:
@dkwon17, question: What do you think, do I need to mention `{prod-short} namespace` here in the overview if I have it mentioned in the secret application procedure?
![image](https://user-images.githubusercontent.com/78564304/161536531-f036054a-8a96-4afd-8d24-fdb7f11df90f.png)

dkwon17 replied:
@max-cx, mentioning {prod-short} namespace here is not absolutely necessary IMO, I suggested it to provide more clarity on the title as to where the Secret is applied.

max-cx replied:
Thank you for clarifying this. In such case, I'll skip it in the overview because `{prod-short} namespace` is not a specific namespace itself.

dkwon17 commented on modules/administration-guide/pages/configuring-authorization-for-private-git-repositories-on-github.adoc:
```diff
@@ -0,0 +1,21 @@
+To enable Che users to access a private Git repository that is hosted on GitHub:
+
+. Set up the GitHub OAuth App
+. Apply the GitHub OAuth App Secret
```
```suggestion
. Apply the GitHub OAuth App Secret on the {prod-short} namespace
```

dkwon17 commented on modules/administration-guide/pages/configuring-authorization-for-private-git-repositories-on-gitlab.adoc:
```diff
@@ -0,0 +1,21 @@
+To enable Che users to access a private Git repository that is hosted using a GitLab instance:
+
+. Set up the GitLab-authorized application
+. Apply the GitLab-authorized application Secret
```
```suggestion
. Apply the GitLab-authorized application Secret on the {prod-short} namespace
```

dkwon17 commented on modules/administration-guide/partials/proc_setting-up-the-bitbucket-application-link.adoc:
```diff
@@ -0,0 +1,100 @@
+
+$ openssl pkcs8 -topk8 -inform pem -outform pem -nocrypt -in private.pem -out privatepkcs8.pem
+
+$ cat privatepkcs8.pem | sed 's/-----BEGIN PRIVATE KEY-----//g' |  sed 's/-----END PRIVATE KEY-----//g' | tr -d '\n' | base64 > privatepkcs8-stripped.pem
```
```suggestion
$ cat privatepkcs8.pem | sed 's/-----BEGIN PRIVATE KEY-----//g' |  sed 's/-----END PRIVATE KEY-----//g' | tr -d '\n' | base64 | tr -d '\n' > privatepkcs8-stripped.pem
```
Since the output of `base64` has newlines, I think it would be great to have an extra `tr -d '\n'` before writing to a file, since the user would have to remove newlines when pasting the content into the secret anyways.

dkwon17 commented on modules/administration-guide/partials/proc_setting-up-the-gitlab-authorized-application.adoc:
```diff
@@ -0,0 +1,52 @@
+[id="setting-up-the-gitlab-authorized-application_{context}"]
+= Setting up the GitLab-authorized application
+
+Set up a GitLab-authorized application as follows:
```
```suggestion
Set up a GitLab-authorized OAuth 2.0 application as follows:
```

max-cx replied:
Thank you. I updated it a bit differently (to keep the GitLab terminology intact). Please see my next commit.

dkwon17 commented on modules/administration-guide/partials/proc_setting-up-the-gitlab-authorized-application.adoc:
```diff
@@ -0,0 +1,52 @@
+// :page-aliases:
+
+[id="setting-up-the-gitlab-authorized-application_{context}"]
+= Setting up the GitLab-authorized application
```
```suggestion
= Setting up the GitLab-authorized OAuth 2.0 application
```

max-cx replied:
Thank you. I updated it elsewhere (to keep the GitLab terminology intact). Please see my next commit.

dkwon17 commented on modules/administration-guide/pages/configuring-authorization-for-private-git-repositories-on-gitlab.adoc:
```diff
@@ -0,0 +1,21 @@
+
+To enable Che users to access a private Git repository that is hosted using a GitLab instance:
+
+. Set up the GitLab-authorized application
```
```suggestion
. Set up the GitLab-authorized OAuth 2.0 application
```

max-cx replied:
Thank you. I updated it elsewhere (to keep the GitLab terminology intact). Please see my next commit.

dkwon17 commented on modules/administration-guide/pages/configuring-authorization-for-private-git-repositories-on-github.adoc:
```diff
@@ -0,0 +1,21 @@
+
+To enable Che users to access a private Git repository that is hosted on GitHub:
+
+. Set up the GitHub OAuth App
```
```suggestion
. Set up the GitHub OAuth 2.0 App
```

max-cx replied:
Thank you. I updated it elsewhere (to keep the GitHub terminology intact). Please see my next commit.

dkwon17 commented on modules/administration-guide/pages/configuring-authorization-for-private-git-repositories-on-bitbucket.adoc:
```diff
@@ -0,0 +1,21 @@
+
+To enable Che users to access a private Git repository that is hosted on a Bitbucket server:
+
+. Set up the Bitbucket application link
```
```suggestion
. Set up the Bitbucket OAuth 1.0 application link
```

max-cx replied:
Thank you. I updated it elsewhere (to keep the Bitbucket terminology intact). Please see my next commit.

dkwon17 commented on modules/administration-guide/partials/proc_setting-up-the-github-oauth-app.adoc:
```diff
@@ -0,0 +1,52 @@
+[id="setting-up-the-github-oauth-app_{context}"]
+= Setting up the GitHub OAuth App
+
+Set up a GitHub OAuth App as follows:
```
```suggestion
Set up a GitHub OAuth 2.0 App as follows:
```

max-cx replied:
Thank you. I updated it a bit differently (to keep the GitHub terminology intact). Please see my next commit.

dkwon17 commented on modules/administration-guide/partials/proc_setting-up-the-github-oauth-app.adoc:
```diff
@@ -0,0 +1,52 @@
+// :page-aliases:
+
+[id="setting-up-the-github-oauth-app_{context}"]
+= Setting up the GitHub OAuth App
```
```suggestion
= Setting up the GitHub OAuth 2.0 App
```

max-cx replied:
Thank you. I updated it elsewhere (to keep the GitHub terminology intact). Please see my next commit.

dkwon17 commented on modules/administration-guide/partials/proc_applying-the-github-oauth-app-secret.adoc:
```diff
@@ -0,0 +1,58 @@
+// :page-aliases:
+
+[id="applying-the-github-oauth-app-secret_{context}"]
+= Applying the GitHub OAuth App Secret
```
```suggestion
= Applying the GitHub OAuth 2.0 App Secret
```

max-cx replied:
Thank you. I updated it elsewhere (to keep the GitHub terminology intact). Please see my next commit.

dkwon17 commented on modules/administration-guide/partials/proc_setting-up-the-bitbucket-application-link.adoc:
```diff
@@ -0,0 +1,100 @@
+// :page-aliases:
+
+[id="setting-up-the-bitbucket-application-link_{context}"]
+= Setting up the Bitbucket application link
```
```suggestion
= Setting up the Bitbucket OAuth 1.0 application link
```

max-cx replied:
Thank you. I updated it elsewhere (to keep the Bitbucket terminology intact). Please see my next commit.

dkwon17 commented on modules/administration-guide/partials/proc_setting-up-the-bitbucket-application-link.adoc:
```diff
@@ -0,0 +1,100 @@
+[id="setting-up-the-bitbucket-application-link_{context}"]
+= Setting up the Bitbucket application link
+
+Set up a Bitbucket application link as follows:
```
```suggestion
Set up a Bitbucket OAuth 1.0 application link as follows:
```

max-cx replied:
Thank you. I updated it a bit differently (to keep the Bitbucket terminology intact). Please see my next commit.

dkwon17 commented on modules/administration-guide/partials/proc_applying-the-bitbucket-application-link-secret.adoc:
```diff
@@ -0,0 +1,63 @@
+Prepare and apply the Bitbucket application link Secret as follows:
+
+.Prerequisites
+* The preceding procedure, Setting up the Bitbucket application link, has been completed.
```
```suggestion
* The preceding procedure, Setting up the Bitbucket OAuth 1.0 application link, has been completed.
```

max-cx replied:
I didn't touch the procedure heading when implementing your suggestion, so this line needn't be updated.

dkwon17 commented on modules/administration-guide/partials/proc_applying-the-github-oauth-app-secret.adoc:
```diff
@@ -0,0 +1,58 @@
+Prepare and apply the GitHub OAuth App Secret as follows:
+
+.Prerequisites
+* The preceding procedure, Setting up the GitHub OAuth App, has been completed.
```
```suggestion
* The preceding procedure, Setting up the GitHub OAuth 2.0 App, has been completed.
```

max-cx replied:
I didn't touch the procedure heading when implementing your suggestion, so this line needn't be updated.

dkwon17 commented on modules/administration-guide/partials/proc_applying-the-gitlab-authorized-application-secret.adoc:
```diff
@@ -0,0 +1,60 @@
+Prepare and apply the GitLab-authorized application Secret as follows:
+
+.Prerequisites
+* The preceding procedure, Setting up the GitLab-authorized application, has been completed.
```
```suggestion
* The preceding procedure, Setting up the GitLab-authorized OAuth 2.0 application, has been completed.
```

max-cx replied:
I didn't touch the procedure heading when implementing your suggestion, so this line needn't be updated.

dkwon17 reviewed (commented):
I think it would be great to specify whether each SCM OAuth app setup is for OAuth 1.0 or OAuth 2.0.

max-cx committed:
adding OAuth 2.0 and 1.0 where approprriate

max-cx committed:
Applying David's suggested change.

Update modules/administration-guide/partials/proc_setting-up-the-bitbucket-application-link.adoc

Co-authored-by: David Kwon <dakwon@redhat.com>

max-cx committed:
deleting an extra space

max-cx committed:
che_fqdn

max-cx committed:
draft

max-cx committed:
draft

jc-berger commented on modules/administration-guide/partials/proc_applying-the-bitbucket-application-link-secret.adoc:
```diff
@@ -0,0 +1,63 @@
+[id="applying-the-bitbucket-application-link-secret_{context}"]
+= Applying the Bitbucket application link Secret
+
+Prepare and apply the Bitbucket application link Secret as follows:
```
I think for the abstract, you don't need the colon to precede the prereqs and procedure. An abstract by nature sets these up. 

So simply, `Prepare and apply the Bitbucket application link Secret.`

max-cx replied:
Thank you, Jake. Done.

jc-berger commented on modules/administration-guide/partials/proc_applying-the-bitbucket-application-link-secret.adoc:
```diff
@@ -0,0 +1,63 @@
+Prepare and apply the Bitbucket application link Secret as follows:
+
+.Prerequisites
+* The preceding procedure, Setting up the Bitbucket application link, has been completed.
```
You don't know if users have looked at the preceding procedure. 

How about simply, `You have set up the Bitbucket application link.` 

You could link to the procedure if you'd like.

max-cx replied:
@jc-berger and @rolfedh, I was under the impression that because of some repository changes such linking wasn't possible in our repository due to Antora at this time (although I used such links in our docs previously). I also realize that mentioning the word `procedure` is leaning toward self-referenced procedures, which is against the SSG. However, I've tried linking it now and links work, at least for the upstream docs (fingers crossed, it will be ok for the downstream docs as well).

I'm still cross-referencing the next/preceding procedure in this PR because both procedures are parts of one super-procedure and have to be performed together, one after the other. If I don't mention the fact that some values come from the preceding procedure, then it's unnecessarily confusing to the reader where to look for those values or how they came into being.

So I've replaced this with the link, and I hope you can review it (I've checked the links):

![image](https://user-images.githubusercontent.com/78564304/161513851-012966b1-8e65-4138-8634-5a884a0d9d20.png)

and I also rewrote the preceding procedure:

![image](https://user-images.githubusercontent.com/78564304/161513750-6608a5ff-5482-4e60-ae4f-00a04c977995.png)

I've edited the rest of the files according to the same structure as above.
If these look okay, Jake, then consider approving this PR.
If there is anything else you think I can improve, by all means let me know here. I'm happy to continue our discussions.

jc-berger replied:
@max-cx yes, I like these rewrites. Thank you for taking my feedback seriously :) 

FWIW, I hear you about links. They're not very encouraged but can make life much easier. I default to user experience, and I agree with you here: if users must complete these procedures in order, then link to the according procedures in prereqs. 

Good work :)

max-cx replied:
Thank you for your support, Jake.

jc-berger commented on modules/administration-guide/partials/proc_applying-the-bitbucket-application-link-secret.adoc:
```diff
@@ -0,0 +1,63 @@
+
+.Prerequisites
+* The preceding procedure, Setting up the Bitbucket application link, has been completed.
+* The following Base64-encoded files that were created during the preceding procedure are prepared:
```
Once again, something link, `You have prepared the following Base64-encoded files:`

jc-berger commented on modules/administration-guide/partials/proc_applying-the-bitbucket-application-link-secret.adoc:
```diff
@@ -0,0 +1,63 @@
+
+.Procedure
+
+. Prepare the secret like so:
```
Minimalism: `Prepare the secret`

max-cx replied:
Thank you, Jake. Done.

jc-berger commented on modules/administration-guide/partials/proc_applying-the-github-oauth-app-secret.adoc:
```diff
@@ -0,0 +1,58 @@
+[id="applying-the-github-oauth-app-secret_{context}"]
+= Applying the GitHub OAuth App Secret
+
+Prepare and apply the GitHub OAuth App Secret as follows:
```
Minimalism: `Prepare and apply the GitHub OAuth App Secret`

max-cx replied:
Thank you, Jake. Done.

jc-berger commented on modules/administration-guide/partials/proc_applying-the-github-oauth-app-secret.adoc:
```diff
@@ -0,0 +1,58 @@
+
+.Procedure
+
+. Prepare the secret like so:
```
`Prepare the secret`

max-cx replied:
Thank you, Jake. Done.

jc-berger commented on modules/administration-guide/partials/proc_applying-the-gitlab-authorized-application-secret.adoc:
```diff
@@ -0,0 +1,60 @@
+[id="applying-the-gitlab-authorized-application-secret_{context}"]
+= Applying the GitLab-authorized application Secret
+
+Prepare and apply the GitLab-authorized application Secret as follows:
```
eliminate `as follows:`

max-cx replied:
Thank you, Jake. Done.

jc-berger commented on modules/administration-guide/partials/proc_applying-the-gitlab-authorized-application-secret.adoc:
```diff
@@ -0,0 +1,60 @@
+Prepare and apply the GitLab-authorized application Secret as follows:
+
+.Prerequisites
+* The preceding procedure, Setting up the GitLab-authorized application, has been completed.
```
Same comment I made earlier about saying `preceding procedure`

jc-berger commented on modules/administration-guide/partials/proc_setting-up-the-bitbucket-application-link.adoc:
```diff
@@ -0,0 +1,100 @@
+[id="setting-up-the-bitbucket-application-link_{context}"]
+= Setting up the Bitbucket application link
+
+Set up a Bitbucket application link using OAuth 1.0 as follows:
```
eliminate `as follows:`

max-cx replied:
Thank you, Jake. Done.

jc-berger commented on modules/administration-guide/partials/proc_setting-up-the-bitbucket-application-link.adoc:
```diff
@@ -0,0 +1,100 @@
+$ openssl rand -base64 24 > bitbucket-consumer-key
+----
+
+. Save the Base64-encoded `bitbucket_consumer_key` file for the next procedure (Applying the Bitbucket application link Secret).
```
Again, I wouldn't reference other procedures as you don't know what other procedures users will go to or complete. 

How about: `Save the Base64-encoded bitbutcket_consumer_key file.`

jc-berger commented on modules/administration-guide/partials/proc_setting-up-the-github-oauth-app.adoc:
```diff
@@ -0,0 +1,52 @@
+[id="setting-up-the-github-oauth-app_{context}"]
+= Setting up the GitHub OAuth App
+
+Set up a GitHub OAuth App using OAuth 2.0 as follows:
```
eliminate `as follows`

max-cx replied:
Thank you, Jake. Done.

jc-berger commented on modules/administration-guide/partials/proc_setting-up-the-github-oauth-app.adoc:
```diff
@@ -0,0 +1,52 @@
+$ echo -n '__<github_oauth_client_id>__' | base64
+----
+
+. Save the Base64-encoded value of *GitHub OAuth Client ID* for the next procedure (Applying the GitHub OAuth App Secret).
```
same comment as above. don't reference other procedures

rolfedh commented on modules/administration-guide/pages/assembly_configuring-oauth-1-for-bitbucket.adoc:
```diff
@@ -0,0 +1,16 @@
+[id="configuring-oauth-1-for-bitbucket_{context}"]
+= Configuring OAuth 1.0 for Bitbucket
+
+To enable Che users to access a private Git repository that is hosted on a Bitbucket server:
```
Do you use a product attribute for "Che?"

max-cx replied:
:) We do of course. What an embarrassment that this slipped through the cracks. But I'm dropping product here altogether as it should be self-explanatory what product I'm referring to here.
```suggestion
To enable users to access a private Git repository that is hosted on a Bitbucket server:
```

rolfedh commented on modules/administration-guide/pages/assembly_configuring-oauth-1-for-bitbucket.adoc:
```diff
@@ -0,0 +1,16 @@
+To enable Che users to access a private Git repository that is hosted on a Bitbucket server:
+
+. Set up the Bitbucket application link (OAuth 1.0)
+. Apply the Bitbucket application link Secret
```
```suggestion
. Set up the Bitbucket application link (OAuth 1.0).
. Apply the Bitbucket application link Secret.
```

rolfedh commented on modules/administration-guide/pages/assembly_configuring-oauth-2-for-github.adoc:
```diff
@@ -0,0 +1,16 @@
+To enable Che users to access a private Git repository that is hosted on GitHub:
+
+. Set up the GitHub OAuth App (OAuth 2.0)
+. Apply the GitHub OAuth App Secret
```
```suggestion
. Set up the GitHub OAuth App (OAuth 2.0).
. Apply the GitHub OAuth App Secret.
```

max-cx replied:
Thank you, Rolfe.

rolfedh commented on modules/administration-guide/pages/assembly_configuring-oauth-2-for-gitlab.adoc:
```diff
@@ -0,0 +1,16 @@
+To enable Che users to access a private Git repository that is hosted using a GitLab instance:
+
+. Set up the GitLab authorized application (OAuth 2.0)
+. Apply the GitLab authorized application Secret
```
```suggestion
. Set up the GitLab authorized application (OAuth 2.0).
. Apply the GitLab authorized application Secret.
```

rolfedh commented on modules/administration-guide/partials/proc_applying-the-bitbucket-application-link-secret.adoc:
```diff
@@ -0,0 +1,63 @@
+  consumer.key: __<base64_content_of_bitbucket_server_consumer_key>__ <4>
+  shared_secret: __<base64_content_of_bitbucket-shared-secret>__ <5>
+----
+<1> The {prod-short} namespace. The default is `{prod-namespace}`.
```
Do your build tools render this instance of `{prod-namespace}` correctly?

max-cx replied:
Yes. Thank you for checking.

max-cx committed:
Update modules/administration-guide/pages/assembly_configuring-oauth-1-for-bitbucket.adoc

Co-authored-by: Rolfe Dlugy-Hegwer <rolfedh@users.noreply.github.com>

max-cx committed:
Update modules/administration-guide/pages/assembly_configuring-oauth-1-for-bitbucket.adoc

Co-authored-by: Rolfe Dlugy-Hegwer <rolfedh@users.noreply.github.com>

max-cx committed:
Update modules/administration-guide/pages/assembly_configuring-oauth-2-for-gitlab.adoc

Co-authored-by: Rolfe Dlugy-Hegwer <rolfedh@users.noreply.github.com>

max-cx committed:
revising the files in response to the language review

max-cx committed:
adding -w0 to base64

max-cx committed:
Revert "adding -w0 to base64"

This reverts commit ed1d2fd4a695b843bab42bfbc4750876a903c3fe.

jc-berger reviewed (approved):
/lgtm

max-cx committed:
draft

max-cx committed:
add orch-cli link

max-cx committed:
correct orch-cli link

max-cx committed:
fix Secret capitalization

max-cx committed:
fix typo

max-cx committed:
fix typos

max-cx committed:
fix

max-cx committed:
draft

rkratky commented on modules/administration-guide/pages/assembly_configuring-oauth-1-for-bitbucket.adoc:
```diff
@@ -0,0 +1,16 @@
+. Set up the Bitbucket application link (OAuth 1.0).
+. Apply the Bitbucket application link Secret.
+
+include::partial$proc_setting-up-the-bitbucket-application-link.adoc[leveloffset=+1]
```
```suggestion
include::partial$proc_setting-up-the-bitbucket-application-link.adoc[leveloffset=+1]

```

rkratky commented on modules/administration-guide/pages/assembly_configuring-oauth-2-for-github.adoc:
```diff
@@ -0,0 +1,16 @@
+. Set up the GitHub OAuth App (OAuth 2.0).
+. Apply the GitHub OAuth App Secret.
+
+include::partial$proc_setting-up-the-github-oauth-app.adoc[leveloffset=+1]
```
```suggestion
include::partial$proc_setting-up-the-github-oauth-app.adoc[leveloffset=+1]

```

rkratky commented on modules/administration-guide/pages/assembly_configuring-oauth-2-for-gitlab.adoc:
```diff
@@ -0,0 +1,16 @@
+. Set up the GitLab authorized application (OAuth 2.0).
+. Apply the GitLab authorized application Secret.
+
+include::partial$proc_setting-up-the-gitlab-authorized-application.adoc[leveloffset=+1]
```
```suggestion
include::partial$proc_setting-up-the-gitlab-authorized-application.adoc[leveloffset=+1]

```

rkratky commented on modules/administration-guide/partials/proc_applying-the-github-oauth-app-secret.adoc:
```diff
@@ -0,0 +1,58 @@
+
+. Apply the Secret by entering it with `{orch-cli}` on a command line:
++
+[source,subs="+quotes,+attributes,+macros"]
```
```suggestion
[subs="+quotes,+attributes,+macros"]
```

rkratky commented on modules/administration-guide/partials/proc_applying-the-gitlab-authorized-application-secret.adoc:
```diff
@@ -0,0 +1,60 @@
+
+. Apply the Secret by entering it with `{orch-cli}` on a command line:
++
+[source,subs="+quotes,+attributes,+macros"]
```
```suggestion
[subs="+quotes,+attributes,+macros"]
```

rkratky replied:
Same thing in all other occurrences of `source` use when it'\s not a code block but an example command line.

max-cx replied:
Thank you. I've adopted this from the old docs. But now that you've mentioned it, I've looked it up--turns out it's an unused attribute: https://docs.asciidoctor.org/asciidoc/latest/verbatim/source-highlighter/#source-language

rkratky commented on modules/administration-guide/partials/proc_setting-up-the-bitbucket-application-link.adoc:
```diff
@@ -0,0 +1,100 @@
+$ openssl rsa -in private.pem -pubout > public.pub
+
+$ cat public.pub | sed 's/-----BEGIN PUBLIC KEY-----//g' | sed 's/-----END PUBLIC KEY-----//g' | tr -d '\n' > public-stripped.pub
+
```
```suggestion
```

rkratky commented on modules/administration-guide/partials/proc_setting-up-the-bitbucket-application-link.adoc:
```diff
@@ -0,0 +1,100 @@
+
+. Save the Base64-encoded `bitbucket-shared-secret` file for use when applying the Bitbucket application link Secret.
+
+. Go to *Administration* *>* *Application Links*
```
```suggestion
. Go to menu:Administration[Application Links].
```
Same thing in other occurrences of menu navigation.

max-cx replied:
Thank you! I'm learning something here.

rkratky reviewed (dismissed):
@max-cx a few nits.

max-cx committed:
Update modules/administration-guide/partials/proc_setting-up-the-bitbucket-application-link.adoc

Co-authored-by: Robert Krátký <rkratky@redhat.com>

max-cx committed:
Update modules/administration-guide/partials/proc_applying-the-gitlab-authorized-application-secret.adoc

Co-authored-by: Robert Krátký <rkratky@redhat.com>

max-cx committed:
Update modules/administration-guide/partials/proc_applying-the-github-oauth-app-secret.adoc

Co-authored-by: Robert Krátký <rkratky@redhat.com>

max-cx committed:
Update modules/administration-guide/partials/proc_setting-up-the-bitbucket-application-link.adoc

Co-authored-by: Robert Krátký <rkratky@redhat.com>

max-cx committed:
revisions in response to Robert,s review

max-cx committed:
Update modules/administration-guide/pages/assembly_configuring-oauth-1-for-bitbucket.adoc

Co-authored-by: Robert Krátký <rkratky@redhat.com>

max-cx committed:
Update modules/administration-guide/pages/assembly_configuring-oauth-2-for-github.adoc

Co-authored-by: Robert Krátký <rkratky@redhat.com>

max-cx committed:
Update modules/administration-guide/pages/assembly_configuring-oauth-2-for-gitlab.adoc

Co-authored-by: Robert Krátký <rkratky@redhat.com>

max-cx committed:
keywords

max-cx committed:
formatting improvements

max-cx committed:
revisions

max-cx committed:
fix

max-cx committed:
last-min change

max-cx committed:
fix

themr0c committed:
Merge branch 'master' into 3792master

max-cx committed:
revision after Mario's review

max-cx committed:
Revert "revision after Mario's review"

This reverts commit 8df5122626d14bc3b63d89201426d73adfcecd81.

max-cx committed:
Revert "Revert "revision after Mario's review""

This reverts commit 34ebffb24e9f08a7d06c155b7b4ef329bf6825f5.

max-cx committed:
fix

themr0c committed:
Merge branch 'master' into 3792master

themr0c commented on modules/administration-guide/partials/proc_setting-up-the-github-oauth-app.adoc:
```diff
@@ -0,0 +1,52 @@
++
+[subs="+quotes,+attributes,+macros"]
+----
+$ echo -n '__<github_oauth_client_id>__' | base64
```
The implementation of the `base64` tool is inconsistent across operating systems. It may add an undesired newline. Therefore it is better to document the conversion to the base64 format using the `openssl` tool, that has consistent behavior.

`$ openssl base64 -e -A `

See: https://www.eclipse.org/che/docs/che-7/end-user-guide/provisioning-the-jetbrains-offline-activation-code/

themr0c replied:
Here, as the output destination is the user (not piped to another tool) it may not be a problem.

max-cx replied:
`echo` has the `-n` option here, so it should be fine.

themr0c commented on modules/administration-guide/partials/proc_setting-up-the-gitlab-authorized-application.adoc:
```diff
@@ -0,0 +1,51 @@
+
+. Click *Save application*.
+
+. Copy the *GitLab Application ID* and encode it to Base64:
```
Consider merging all remaining step into one single step:

* Convert *GitLab Application ID* and *GitLab Client Secret* values to the Base64 format for later use in {prod-short}.

max-cx replied:
Thanks for the suggestion.
I've merged two steps, yet differently:

![image](https://user-images.githubusercontent.com/78564304/163005442-a5214dea-f87e-43a9-84fd-7defd993540d.png)

themr0c commented on modules/administration-guide/partials/proc_setting-up-the-github-oauth-app.adoc:
```diff
@@ -0,0 +1,52 @@
+
+. Click *Generate new client secret*.
+
+. Copy the *GitHub OAuth Client ID* and encode it to Base64:
```
Consider merging all remaining step into one single step:

* Convert *GitHub OAuth Client ID* and *GitHub OAuth Client Secret* values to the Base64 format for later use in {prod-short}.

max-cx replied:
Thanks for the suggestion.
I've merged two steps, yet differently:

![image](https://user-images.githubusercontent.com/78564304/163004925-617e8d0c-836e-4f51-8ca6-f944b03a5b24.png)

themr0c commented on modules/administration-guide/partials/proc_setting-up-the-bitbucket-application-link.adoc:
```diff
@@ -0,0 +1,65 @@
+. Enter *{prod-short}* as the *Consumer name*.
+
+. Paste the content of the `public-stripped.pub` file as the *Public Key* and click *Continue*.
+
```
In comparison to other providers, a last step is missing:

* Convert *privatepkcs8-stripped.pem*, *bitbucket_consumer_key*, and *bitbucket-shared-secret* values to the Base64 format for later use in {prod-short}.

themr0c replied:
I apoligize, it's in step 1.

themr0c commented on modules/administration-guide/partials/proc_setting-up-the-github-oauth-app.adoc:
```diff
@@ -0,0 +1,52 @@
+
+. Enter `pass:c,a,q[{prod-url}]/api/oauth/callback` as the *Authorization callback URL*.
+
+. Click *Register application*.
```
Consider concatenating in one single step:

. Go to link:https://github.com/settings/applications/new[the Register a new OAuth application form], fill the form, and click *Register application*. Use following values:

(here a key / value table)

max-cx replied:
Thanks for the suggestion.
Unfortunately, this approach doesn't seem to work for single-line steps.
Currently, I have only three single-line steps.
If grouping these steps into one, I end up with one additional line of text even though the step count is reduced:
![image](https://user-images.githubusercontent.com/78564304/162999360-62893377-7a27-46d2-8fe8-b098b7787068.png)

themr0c commented on modules/administration-guide/partials/proc_setting-up-the-github-oauth-app.adoc:
```diff
@@ -0,0 +1,52 @@
+
+. Click *Register application*.
+
+. Click *Generate new client secret*.
```
```suggestion
. Click *Generate a new client secret*.
```

max-cx replied:
This is GitHub UI. I can't make changes to their labels.

max-cx committed:
revisions after Fabrice's review
